src/components/compositions/notifications/index.ts

```diff
@@ -0,0 +1,154 @@
+import { computed, ref, watch } from "@vue/composition-api";
+import { web3 } from "@/api/web3";
+
+const maxLimit = 100;
+const defaultShowSeconds = 7;
+
+export enum ENotificationStatus {
+  success = "success",
+  info = "info",
+  warning = "warning",
+  error = "error",
+  pending = "pending"
+}
+
+export interface INotificationView {
+  id: number;
+  status: ENotificationStatus;
+  title: string;
+  description: string;
+  txHash?: string;
+  timestamp: Date;
+  showSeconds?: number;
+  alertOnly?: boolean;
+}
+export interface IAddNotification {
+  title: string;
+  description: string;
+  status?: ENotificationStatus;
+  txHash?: string;
+  showSeconds?: number;
+  alertOnly?: boolean;
+}
+
+export const history = ref<INotificationView[]>([]);
+
+export const alertQueue = computed(() =>
+  history.value.filter(n => n.showSeconds).reverse()
+);
+
+export const pendingQueue = computed(() =>
+  history.value.filter(n => n.status === ENotificationStatus.pending)
+);
```

Comment:
Why is it that we're ```.reverse()``` in the ```alertQueue``` but not the ```pendingQueue``` ?

Reply:
Because we want the oldest alert to be at the top since it is the first Element to disappear after 7s.

For pending history on the other hand, we want the most recent notification at the top.